jest-enzyme typescript declaration no longer working in 6.0.3

cc @astorije

As of 6.0.3, I started getting compile errors

```
components/ui/LineClamp.test.tsx:91:38 - error TS2339: Property 'toHaveProp' does not exist on type 'Matchers<ReactWrapper<HTMLAttributes, any, Component<{}, {}, any>>>'.

91     expect(wrapper.find("Truncate")).toHaveProp("lines", false);
                                        ~~~~~~~~~~


components/ui/Loading/Loading.test.tsx:14:48 - error TS2339: Property 'toExist' does not exist on type 'Matchers<ReactWrapper<HTMLAttributes, any, Component<{}, {}, any>>>'.

14     expect(wrapper.find("[aria-valuenow=20]")).toExist();
                                                  ~~~~~~~
```

It happens due to #245 which adds an import statement to index.d.ts. This turns index.d.ts into a regular module and breaks the ambient declaration of the jest namespace.

See: https://github.com/Microsoft/TypeScript/issues/5506#issuecomment-153254151

I modified the declaration file to look like this:

```ts
/// <reference types="react" />

// import * as Enzyme from 'enzyme';
// ^^^^^^ this is the problem
// declare global {
//    var shallow: typeof Enzyme.shallow;
//    var mount: typeof Enzyme.mount;
//    var render: typeof Enzyme.render;
// }

declare namespace jest {
    interface Matchers<R> {
        toBeChecked(): void;
        toBeDisabled(): void;
        toBeEmptyRender(): void;
        toContainReact(component: React.ReactElement<any>): void;
        toExist(): void;
        toHaveClassName(className: string): void;
        toHaveHTML(html: string): void;
        toHaveProp(propKey: object|string, propValue?: any): void;
        toHaveRef(refName: string): void;
        toHaveState(stateKey: object|string, stateValue?: any): void;
        toHaveStyle(styleKey: object|string, styleValue?: any): void;
        toHaveTagName(tagName: string): void;
        toHaveText(text: string): void;
        toHaveValue(value: any): void;
        toIncludeText(text: string): void;
        toMatchElement(element: React.ReactElement<any>): void;
        toMatchSelector(selector: string): void;
    }
}
```
... and things are back to working

I understand this is effectively reverting #245 but I suspect a bunch of users will face this issue upon upgrading.
It also seems that `jest-enzyme` is not the right place to be declaring these globals since this module is not actually setting `shallow`, `mount`, `render` on `global`. They are being set by `jest-environment-enzyme`. Perhaps that module can have a declaration file to add the three functions on to the global namespace

--- a/packages/jest-environment-enzyme/package.json
+++ b/packages/jest-environment-enzyme/package.json
@@ -16,6 +16,7 @@
   "files": [
     "lib"
   ],
+  "types": "lib/index.d.ts",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/FormidableLabs/enzyme-matchers.git"

--- a/packages/jest-environment-enzyme/src/index.d.ts
+++ b/packages/jest-environment-enzyme/src/index.d.ts
@@ -1,0 +1,7 @@
+import * as Enzyme from 'enzyme';
+
+declare global {
+    var shallow: typeof Enzyme.shallow;
+    var mount: typeof Enzyme.mount;
+    var render: typeof Enzyme.render;
+}

--- a/packages/jest-enzyme/src/index.d.ts
+++ b/packages/jest-enzyme/src/index.d.ts
@@ -1,13 +1,5 @@
 /// <reference types="react" />
 
-import * as Enzyme from 'enzyme';
-
-declare global {
-    var shallow: typeof Enzyme.shallow;
-    var mount: typeof Enzyme.mount;
-    var render: typeof Enzyme.render;
-}
-
 declare namespace jest {
     interface Matchers<R> {
         toBeChecked(): void;

